src/types/dict.rs

```diff
@@ -148,13 +148,19 @@ impl PyDict {
     /// Note that it's unsafe to use when the dictionary might be changed
     /// by other python code.
     pub fn iter(&self) -> PyDictIterator {
-        PyDictIterator { dict: self, pos: 0 }
+        let py = self.py();
+        PyDictIterator {
+            dict: self.to_object(py),
+            pos: 0,
+            __marker: PhantomData,
+        }
     }
 }
 
-pub struct PyDictIterator<'a> {
-    dict: &'a PyDict,
+pub struct PyDictIterator<'dict> {
+    dict: PyObject,
     pos: isize,
+    __marker: PhantomData<&'dict ()>,
```

Comment:
Is there a special reason not to use `Python` as marker here. Also imho we don't need the underscores since the fields are private anyway